ouroboros-consensus/src/Ouroboros/Consensus/Util/IOLike.hs

```diff
@@ -91,3 +93,20 @@ class ( MonadAsync              m
       ) => IOLike m where
 
 instance IOLike IO
+
+{-------------------------------------------------------------------------------
+  MonadDelay Wrappers
+-------------------------------------------------------------------------------}
+
+-- | This can be used to sleep for longer periods of time. In some cases,
+-- 'threadDelay' can't be used for delays over ~35mins, because it uses an
+-- 'Int', which can overflow. See 'diffTimeToMicrosecondsAsInt' for more info.
+threadSleep :: MonadDelay m => DiffTime -> m ()
+threadSleep time
```

Comment:
An even better idea from @edsko: call this one `threadDelay` so that all other changes in this PR are no longer needed and that we reduce the chances of accidentally importing/using the broken `threadDelay` (typically you import `IOLike` too, so you'd have a conflict).